### PR TITLE
Add simplified tuning definition

### PR DESCRIPTION
**Pull request by js-jslog**

### CODE REVIEW

File: packages/harpguru-core/src/components/notify-of-quiz-question/utils/get-counterpart-degree-id/get-counterpart-degree-id.ts [hunk 1]
```diff
@@ -12,7 +12,8 @@ export const getCounterpartDegreeId = (props: Props): DegreeIds => {
   const { pitchId, harpKeyId, pozitionId } = props
 
   const surrogateHarpStrata = getHarpStrata({
-    apparatusId: getApparatusIds()[0],
+    tuningId: getTuningIds()[0],
+    valvingId: ValvingIds.NotValved,
```
Comment (js-jslog) [inline packages/harpguru-core/src/components/notify-of-quiz-question/utils/get-counterpart-degree-id/get-counterpart-degree-id.ts]: Could change this to `getValvingIds()[0]` for consistency and reliability.

File: packages/harpparts/src/apparatus/utils/build-apparatus/index.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { reedArrayToMatrices } from '../reed-array-to-matrices'
+import type { Apparatus } from '../../types'
+import { ValvingIds } from '../../../valving'
+import type { TuningIds } from '../../../tuning'
+import { getTuning } from '../../../access-parts'
+
+export const buildApparatus = (
+  tuningId: TuningIds,
+  valvingId: ValvingIds = ValvingIds.NotValved
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/build-apparatus/index.ts]: Do we want to leave this as a non-mandatory field? I don't think so, and I don't think that the effort of updating all the references is a good reason not to.

File: packages/harpparts/src/apparatus/utils/build-apparatus/index.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+): Apparatus => {
+  const tuning = getTuning(tuningId)
+  return {
+    tuningId: tuning.id,
+    valvingId,
+    ...reedArrayToMatrices(
+      tuning.reedArray,
+      tuning.id,
+      valvingId === ValvingIds.HalfValved
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/build-apparatus/index.ts]: I think the `reedArrayToMatrices` function should probably have it's parameter changed from a `boolean` to a `ValvingIds`.

File: packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.test.ts [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+    draw: 4,
+    blowbends: [],
+    drawbends: [],
+    overblows: [],
+    overdraws: [],
+    valvedblows: [],
+    valveddraws: [],
+  }
+  expect(isHoleValid({ ...hole, drawbends: [2, 1, 0] })).toStrictEqual([
+    HoleErrors.NonconsecutiveBends,
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.test.ts]: The names and capitalisations in the names of `HoleErrors` enum need to be reviewed.

File: packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import type { Hole } from '../../types'
+import {
+  isConsecutiveWithPrevious,
+  Direction,
+} from '../../../packages/is-consecutive-with-previous'
+
+export enum HoleErrors {
+  ConflictingDrawBends = 'CONFLICTING_DRAW_BENDS',
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts]: Should be `ConflictingDrawbends` for consistency.

Also the next one should be `ConflictingBlowbends`.

File: packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  NonconsecutiveBends = 'NONCONSECUTIVE_BENDS',
+  NonconsecutiveBlowbends = 'NONCONSECUTIVE_BLOWBENDS',
+  NonconsecutiveOverblows = 'NONCONSECUTIVE_OVERBLOWS',
+  NonconsecutiveOverdraws = 'NONCONSECUTIVE_OVERDRAWS',
+  ConflictingValvedBlowBends = 'CONFLICTING_VALVED_BLOW_BENDS',
+  ConflictingValvedDrawBends = 'CONFLICTING_VALVED_DRAW_BENDS',
+}
+
+export const isHoleValid = (hole: Hole): HoleErrors[] => {
+  const bendLimit = 5
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts:32]: Add a comment here that this limit is actually more of a comment on the sensible limit of reed  distance. Bends won't really be possible at all if the reeds are far enough apart to be attempting to produce 6 semitone bends.

Perhaps this limit should be discovered more directly by just counting the distance between the reeds. That's more direct, but is it as semantic? I think not.

File: packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  const overbendLimit = 2
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts:33]: We should have a limit in here for valved bends too.

File: packages/harpparts/src/apparatus/utils/map-hole-to-filter-if-valved/map-hole-to-filter-if-valved.test.ts [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+  const output1 = mapHoleToFilterIfValved(input1)
+  expect(output1).toStrictEqual({ ...input1, overblows: [] })
+
+  // This captures a rare event where the lower reed is valved. If this scenario occured
+  // then this is how we would expect it to behave. But it's unlikely that anyone would
+  // want to do this since you end up with a worse blow bend than you would without the
+  // valving. We may even add some error checking in the hole validator which throws an
+  // error if this setup is detected. But here is not the place for presumptions of that
+  // kind. Here we'll just work logically.
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/map-hole-to-filter-if-valved/map-hole-to-filter-if-valved.test.ts]: Should we consider the mentioned validation or is it too early for that?

File: packages/harpparts/src/apparatus/utils/map-hole-to-include-bends/map-hole-to-include-bends.test.ts [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+    valvedblows: [],
+    valveddraws: [],
+  }
+
+  const actualHoleOutput = mapHoleToIncludeBends(holeInput)
+
+  expect(actualHoleOutput).toStrictEqual(expectedHoleOutput)
+})
+
+test('mapHoleToIncludeBends adds no blow bends', () => {
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/map-hole-to-include-bends/map-hole-to-include-bends.test.ts]: "..when reeds are 1 semitone apart".

File: packages/harpparts/src/apparatus/utils/map-hole-to-include-bends/map-hole-to-include-bends.test.ts [hunk 10]
```diff
@@ -0,0 +1,162 @@
+import type { Hole } from '../../types'
+
+import { mapHoleToIncludeBends } from './map-hole-to-include-bends'
+
+test('mapHoleToIncludeBends adds no bends', () => {
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/map-hole-to-include-bends/map-hole-to-include-bends.test.ts]: "..when reeds are 1 semitone apart".

File: packages/harpparts/src/apparatus/utils/map-hole-to-include-overbends/map-hole-to-include-overbends.test.ts [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    ...holeInput,
+    overdraws: [2],
+  }
+  const actualHoleOutput = mapHoleToIncludeOverbends(holeInput)
+
+  expect(actualHoleOutput).toStrictEqual(expectedHoleOutput)
+})
+
+// This behaviour may not actually be correct, but I'm waiting for Woozle to do
+// an experiment to confirm whether overbends occur in this situation.
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/map-hole-to-include-overbends/map-hole-to-include-overbends.test.ts]: It's time to confirm this.

File: packages/harpparts/src/apparatus/utils/map-hole-to-include-valvebends/map-hole-to-include-valvebends.test.ts [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+    valveddraws: [8],
+  }
+  const actualHoleOutput = mapHoleToIncludeValvebends(holeInput)
+
+  expect(actualHoleOutput).toStrictEqual(expectedHoleOutput)
+})
+
+// The way I see it, if you're halfvalving normally but your hole would have no bends
+// for the unvalved reed then you might as well go full valved.
+// This is an edge case anyway.
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/map-hole-to-include-valvebends/map-hole-to-include-valvebends.test.ts]: This needs rewording a bit, otherwise it would seem to justify going full valved on holes where the reeds are only a semitone apart. It doesn't because actually, while there isn't a full bend there there is a microtonal one which *is* useful and probably preferable to valved bends in most cases because it's kinder on the reed and easier to control the amplitude.

File: packages/harpparts/src/apparatus/utils/reed-array-to-matrices/reed-array-to-matrices.ts [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+  reedArray: ReedArray,
+  tuningId: TuningIds,
+  isHalfValved?: boolean
+): Pick<Apparatus, 'halfstepIndexMatrix' | 'interactionMatrix'> => {
+  const holeArray = pivotReedArray(reedArray)
+    .map(mapReedPairToHole)
+    .map(mapHoleToIncludeBends)
+    .map(mapHoleToIncludeOverbends)
+    .map(mapHoleToFilterOverbends)
+    .map((hole) => (isHalfValved ? mapHoleToIncludeValvebends(hole) : hole))
```
Comment (js-jslog) [inline packages/harpparts/src/apparatus/utils/reed-array-to-matrices/reed-array-to-matrices.ts]: Should the `mapHoleToIncludeValvebends` function actually receive the valving type and make the decision about whether to modify the `Hole` or not?

File: packages/harpparts/src/tuning/constants/index.ts [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+export const POWER_DRAW_TUNING: Tuning = {
+  id: TuningIds.PowerDraw,
+  // prettier-ignore
+  reedArray: [
+    // 1    2    3    4    5    6    7    8    9   10
+    [ c1 , e1 , g1 , c2 , e2 , g2 , a2 , c3 , e3 , a3 ],
+    [ d1 , g1 , b1 , d2 , f2 , a2 , b2 , d3 , g3 , c4 ],
+  ], // TODO: How do we define ReedArray as a non-mutable type. I can't make this `as const` for some reason.
```
Comment (js-jslog) [inline packages/harpparts/src/tuning/constants/index.ts]: TODO

File: packages/harpstrata/src/get-harp-strata/utils/get-degree-matrix/get-degree-matrix.test.ts [hunk 15]
```diff
@@ -20,7 +20,7 @@ test('getDegreeMatrix function maps a simple 2d array of 0s to 4th degrees (6) w
 test('getDegreeMatrix maps a major diatonic halfstepmatrix in to a major diatonic degreematrix in first pozition', () => {
   const { MAJOR_DIATONIC_FIRST_POZITION } = EXAMPLE_DEGREE_MATRICES
   const actualArray = getDegreeMatrix(
-    MAJOR_DIATONIC_APPARATUS.halfstepIndexMatrix,
+    majorDiatonicApparatus.halfstepIndexMatrix,
```
Comment (js-jslog) [inline packages/harpstrata/src/get-harp-strata/utils/get-degree-matrix/get-degree-matrix.test.ts:38]: Need to double-check whether there are other places where this decapitalisation should happen.

**Review by js-jslog — COMMENTED** [review]
Mustn't forget to update the CHANGELOGs of all of the packages.

File: packages/harpparts/src/apparatus-integration.test.ts [hunk 16]
```diff
@@ -0,0 +1,59 @@
+import type { TuningIds } from './tuning'
+import type { Interaction } from './interaction'
+import { buildApparatus } from './apparatus'
+import { getTuningIds } from './access-parts'
+
+import type { HalfstepIndex, HarpFaceMatrix, HarpFaceRow } from './types'
+
+// TODO: the parity checking logic in here needs it's own test to make sure it's working.
```
[the remark this answers is not in the document]
Comment (js-jslog) [reply packages/harpparts/src/apparatus-integration.test.ts]: Done

File: packages/harpparts/src/apparatus/types/index.ts [hunk 17]
```diff
@@ -1,36 +1,53 @@
...
+  ReedPair,
+  ReedPair
+]
 
-export type Interaction = {
-  readonly id: InteractionIds
+export type Hole = {
+  readonly blow: HalfstepIndex
+  readonly draw: HalfstepIndex
+  readonly bends: HalfstepIndex[]
```
[the remark this answers is not in the document]
Comment (js-jslog) [reply packages/harpparts/src/apparatus/types/index.ts]: Done

File: packages/harpguru-core/src/components/menu-of-covariants/utils/get-new-harpstrata-for-dispatcher/get-new-harpstrata-for-dispatcher.test.ts [hunk 18]
```diff
@@ -1,12 +1,13 @@
 import { getHarpStrata } from 'harpstrata'
-import { ApparatusIds, PitchIds, PozitionIds } from 'harpparts'
+import { TuningIds, PitchIds, PozitionIds, ValvingIds } from 'harpparts'
 
 import type { GlobalState } from '../../../../types'
 
 import { getNewHarpStrataForDispatcher } from './get-new-harpstrata-for-dispatcher'
```
[the remark this answers is not in the document]
Comment (js-jslog) [reply packages/harpguru-core/src/components/menu-of-covariants/utils/get-new-harpstrata-for-dispatcher/get-new-harpstrata-for-dispatcher.test.ts]: Done

File: packages/harpguru-core/src/components/menu-of-settings/hooks/use-valving-items/use-valving-items.tsx [hunk 19]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { getValvingIds, ValvingIds } from 'harpparts'
+
+import { OptionItem } from '../../../option-item'
+import type { OptionItemProps } from '../../../option-item'
+import type { UseGlobal } from '../../../../types'
+
+type ItemCallback = ValvingIds
```
[the remark this answers is not in the document]
Comment (js-jslog) [reply packages/harpguru-core/src/components/menu-of-settings/hooks/use-valving-items/use-valving-items.tsx:8]: Yes it is. Perhaps this type could be better named, but it's not so bad that I want to commit the time now or as techdebt.

File: packages/harpguru-core/src/components/notify-of-quiz-question/utils/get-counterpart-degree-id/get-counterpart-degree-id.ts (hunk 1 above)
Comment (js-jslog) [reply packages/harpguru-core/src/components/notify-of-quiz-question/utils/get-counterpart-degree-id/get-counterpart-degree-id.ts]: Done

File: packages/harpparts/src/apparatus/utils/build-apparatus/index.ts (hunk 2 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/build-apparatus/index.ts]: Done

File: packages/harpparts/src/apparatus/utils/build-apparatus/index.ts (hunk 3 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/build-apparatus/index.ts]: Done

File: packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.test.ts (hunk 4 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.test.ts]: Done

File: packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts (hunk 5 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts]: Done

File: packages/harpparts/src/apparatus/utils/map-hole-to-include-bends/map-hole-to-include-bends.test.ts (hunk 9 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/map-hole-to-include-bends/map-hole-to-include-bends.test.ts]: Done

File: packages/harpparts/src/apparatus/utils/map-hole-to-include-bends/map-hole-to-include-bends.test.ts (hunk 10 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/map-hole-to-include-bends/map-hole-to-include-bends.test.ts]: Done

File: packages/harpparts/src/apparatus/utils/reed-array-to-matrices/reed-array-to-matrices.ts (hunk 13 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/reed-array-to-matrices/reed-array-to-matrices.ts]: Done

File: packages/harpparts/src/apparatus/utils/map-hole-to-filter-if-valved/map-hole-to-filter-if-valved.test.ts (hunk 8 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/map-hole-to-filter-if-valved/map-hole-to-filter-if-valved.test.ts]: Function removed

File: packages/harpparts/src/apparatus/utils/map-hole-to-include-overbends/map-hole-to-include-overbends.test.ts (hunk 11 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/map-hole-to-include-overbends/map-hole-to-include-overbends.test.ts]: Done

File: packages/harpparts/src/apparatus/utils/map-hole-to-include-valvebends/map-hole-to-include-valvebends.test.ts (hunk 12 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/map-hole-to-include-valvebends/map-hole-to-include-valvebends.test.ts]: Done

File: packages/harpparts/src/tuning/constants/index.ts (hunk 14 above)
Comment (js-jslog) [reply packages/harpparts/src/tuning/constants/index.ts]: Done

File: packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts (hunk 6 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts:32]: Done

File: packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts (hunk 7 above)
Comment (js-jslog) [reply packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.ts:33]: Done

File: packages/harpstrata/src/get-harp-strata/utils/get-degree-matrix/get-degree-matrix.test.ts (hunk 15 above)
Comment (js-jslog) [reply packages/harpstrata/src/get-harp-strata/utils/get-degree-matrix/get-degree-matrix.test.ts:38]: I've not been following a well articulated convention. I think my opinion was basically that test data in particular deserves capitalisation to demonstrate that it is explicitly stated at it's source, but the edges of how I think about it are hazy.

I don't think I'll be able to remove all imprecision in my utilisation of capitalisation but I think airbnb's definition can support me here: https://github.com/airbnb/javascript/#naming--uppercase

Comment (js-jslog) [reply packages/harpstrata/src/get-harp-strata/utils/get-degree-matrix/get-degree-matrix.test.ts:38]: Much improved

File: packages/harpparts/src/access-parts/get-harp-part/get-tuning.ts [hunk 20]
```diff
@@ -0,0 +1,8 @@
+import { ORDERED_TUNINGS } from '../constants'
+import type { TuningIds, Tuning } from '../../tuning'
+
+export const getTuning = (tuningId: TuningIds): Tuning => {
+  const tuning = ORDERED_TUNINGS.get(tuningId)
+  if (tuning === undefined) throw 'A tuning id for an unlisted tuning was used'
```
[the remark this answers is not in the document]
Comment (js-jslog) [reply packages/harpparts/src/access-parts/get-harp-part/get-tuning.ts:6]: No... There is no convenient way I can find to do this. This issue has been created to consider this at a later date: https://github.com/js-jslog/harpguru/issues/100

**Review by js-jslog — COMMENTED** [review]
LGTM
